### PR TITLE
Add linalg_vector_norm to core decompositions

### LINKED ISSUE #150061: [Export] [Core ATen] [Decomposition] `linalg_vector_norm` not decomposed
### 🐛 Describe the bug

As defined in [native functions](https://github.com/pytorch/pytorch/blob/v2.6.0/aten/src/ATen/native/native_functions.yaml#L14189-L14199), operator `linalg_vector_norm` is not tagged as `core`, which means it doesn't belong to core aten. In another word, when we run
```
import torch


class Model(torch.nn.Module):
    def forward(self, x):
        return torch.linalg.vector_norm(x)


model = Model()
model.eval()

x = torch.rand(2, 3)
aten_program = torch.export.export(model, (x,))
print(aten_program)

core_aten_program = aten_program.run_decompositions()
print(core_aten_program)
```
We expect to see it decomposed in `core_aten_program`. But actually, now it isn't

### Versions

torch 2.6.0

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* __->__ #185735

`aten.linalg_vector_norm` already had a refs decomposition, but it was not
included in the default core ATen decomposition table used by
ExportedProgram.run_decompositions(). As a result, exported programs kept
`aten.linalg_vector_norm.default` even though the op is not part of the core
ATen opset.

A previous table-only attempt exposed a correctness failure in the decomposition
cross-reference test harness rather than in the forward decomposition. The
harness identified mutable ops with `func.name()[-1] == "_"`, which misses
overloaded mutable schemas such as `aten::pow_.Scalar`. During the
linalg_vector_norm backward check this decomposed an in-place `pow_` from
`norm_backward` and then ran the real op for comparison, double-mutating the
intermediate. This patch uses schema mutability to skip all mutable overloads in
that harness.

While validating the refs decomposition, I also fixed the `dim=()` path for
non-special norms by canonicalizing empty dim sequences before constructing the
reduction and singleton fast path. That makes the decomposition match native
shape behavior when `keepdim=True`.

The main path is to add the existing refs decomposition to the core table rather
than creating a new duplicate decomposition, because the registered refs formula
already models the operator and now passes the relevant forward and backward
coverage.

Fixes #150061
Generated by my agent

Test Plan:
- python test/test_decomp.py TestDecompCPU.test_decomp_crossref_mode_skips_overloaded_inplace_cpu -v
- python test/test_decomp.py TestDecompCPU.test_quick_core_backward_linalg_vector_norm_cpu_float64 -v
- python test/test_decomp.py DecompOneOffTestsCPU.test_linalg_vector_norm_decomp_correctness_cpu -v
- python test/test_decomp.py HasDecompTest.test_aten_core_operators -v
- python test/export/test_export.py TestExport.test_export_decomps_linalg_vector_norm -v
- lintrunner -a
- git diff --check

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I fixed the Dr. CI test_decomp failures by narrowing the DecompCrossRefMode in-place skip to operator base names ending in '_' instead of all mutable schemas, so mutable functional ops like _batch_norm_with_update still get decomposed while overloaded in-place ops like pow_.Scalar are skipped. I also confirmed the docs failure was aws: command not found in the docs upload step and not from files touched by this PR.

**Comment by jansel** [comment]
My agent says I fixed the latest Dr. CI test_decomp failures: addcdiv and lerp were now passing after the DecompCrossRefMode in-place overload fix, so their stale core_backward_failures xfails were causing unexpected-success failures. I removed those two stale xfails and verified both targeted tests pass locally.

**Comment by jansel** [comment]
My agent says I fixed the latest Dr. CI test_decomp failures: binary_cross_entropy_with_logits is now passing after the DecompCrossRefMode in-place overload fix, so its stale core_backward_failures xfail was causing unexpected-success failures. I removed that stale xfail and verified the targeted test passes locally with the current checkout's stale torch._C._remove_obj_from_tls extension symbol shim.

File: test/test_decomp.py [hunk 1]
```diff
@@ -979,6 +976,19 @@ def check_decomposed(self, aten_name, mode):
...
+        x = torch.tensor(2.0, device=device)
+        mode = self.DecompCrossRefMode(
+            self, self.precision, self.rel_tol, x.dtype, run_all=False
+        )
+        with mode, enable_python_dispatcher():
+            aten.pow_.Scalar(x, 3.0)
+
+        self.assertIn(aten.pow_.Scalar, mode.called)
+        self.assertNotIn(aten.pow_.Scalar, mode.decomposed)
+
```
Comment (aorenste) [inline test/test_decomp.py:993]: [cleanup] This locks in only the skip direction (an in-place overload is not decomposed). The reason the fix uses a base-name `_` suffix rather than `func._schema.is_mutable` is that mutable *functional* ops must still be decomposed -- per the PR notes, `_batch_norm_with_update` is the motivating case. Nothing here guards that direction: a later switch to a broader mutability predicate would still pass this test while silently dropping functional-mutable decomposition coverage (the same class the removed addcdiv / lerp / binary_cross_entropy_with_logits xfails relied on). Consider also asserting that a mutable-but-functional op IS in `mode.decomposed`.

**Review by aorenste — APPROVED** [review]
(Reviewed by me, assisted by AI)

[cleanup] The description's harness paragraph is stale. It says "This patch uses schema mutability to skip all mutable overloads in that harness," but the shipped code is `in_place = func._schema.name.endswith("_")` -- a base-name suffix check, not mutability. The two diverge exactly on mutable *functional* ops (e.g. `_batch_norm_with_update`), which the shipped logic intentionally keeps decomposing; `is_mutable` was the earlier approach (the one the `@claude` review comment looked at) and was dropped for that reason. Since this is ghstack, the description becomes the squashed commit / PR message -- please update it to describe the base-name check so the committed record matches the code.

File: test/test_decomp.py (hunk 1 above)
Comment (jansel) [reply test/test_decomp.py:993]: My agent says I added a companion regression for this direction: `test_decomp_crossref_mode_decomposes_mutable_functional_ops` now calls `aten._batch_norm_with_update.default` and asserts that the mutable functional op is present in `mode.decomposed`, while the existing `pow_.Scalar` test still verifies overloaded in-place ops are skipped.

**Comment by jansel** [comment]
My agent says I updated the ghstack commit/PR description to describe the schema base-name `_` check rather than schema mutability, and noted that the tests now cover both overloaded in-place skipping and mutable functional decomposition.

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by jansel** [comment]
My agent says the merge attempt was blocked by a cancelled inductor_cpp_wrapper CI job, while Dr. CI classifies the remaining failures as flaky or broken trunk and unrelated to this PR. I reran the failed inductor workflow jobs so CI can refresh.
